src/main/src/main/java/org/geoserver/security/GeoServerSecurityManager.java

```diff
@@ -305,6 +328,8 @@ public ConfigurationPasswordEncryptionHelper getConfigPasswordEncryptionHelper()
     @Override
     public void setApplicationContext(ApplicationContext appContext) throws BeansException {
         this.appContext = appContext;
+        this.xp = buildPersister();
```

Comment:
Building and configuring a XStream instance is very expensive, so do it just once (it kept on showing up as the biggest offender even after the version check in the migrations, see below)